internal/http/healthcheck_server_test.go

```
@@ -0,0 +1,166 @@
+					t.Errorf("handleHealthCheck() Content-Type = %v, want %v", contentType, "application/json")
+				}
+			}
+		})
+	}
+}
+
+func TestHealthCheckHandlerEncodingError(t *testing.T) {
+	req := httptest.NewRequest(http.MethodGet, "/healthz", nil)
+
+	failingWriter := &failingWriter{
+		headers: http.Header{},
+	}
+	handleHealthCheck(failingWriter, req)
+
+	if failingWriter.statusCode != http.StatusInternalServerError {
+		t.Errorf("handleHealthCheck() with encoding error, status = %v, want %v",
+			failingWriter.statusCode, http.StatusInternalServerError)
+	}
+}
+
+type failingWriter struct {
+	statusCode int
+	headers    http.Header
+}
+
+func (w *failingWriter) Header() http.Header {
+	return w.headers
+}
+
+func (w *failingWriter) Write([]byte) (int, error) {
+	return 0, fmt.Errorf("encoding error")
+}
+
+func (w *failingWriter) WriteHeader(statusCode int) {
+	w.statusCode = statusCode
+}
+
+func TestNewHealthCheckServer(t *testing.T) {
+	server := NewHealthCheckServer()
+
+	if server == nil {
+		t.Fatal("NewHealthCheckServer() returned nil")
+		return
+	}
+
+	if server.Addr != ":5680" {
+		t.Errorf("NewHealthCheckServer() addr = %v, want %v", server.Addr, ":5680")
+	}
+
+	if server.ReadTimeout != readTimeout {
+		t.Errorf("NewHealthCheckServer() readTimeout = %v, want %v", server.ReadTimeout, readTimeout)
+	}
+
+	if server.WriteTimeout != writeTimeout {
+		t.Errorf("NewHealthCheckServer() writeTimeout = %v, want %v", server.WriteTimeout, writeTimeout)
+	}
+}
+
+func TestGetPort(t *testing.T) {
+	tests := []struct {
+		name         string
+		envPort      string
+		expectedPort int
+	}{
+		{
+			name:         "returns default port when env var is not set",
+			envPort:      "",
+			expectedPort: defaultPort,
+		},
+		{
+			name:         "returns custom port when valid env var is set",
+			envPort:      "8080",
+			expectedPort: 8080,
+		},
+		{
+			name:         "returns default port when env var is negative",
+			envPort:      "-1",
+			expectedPort: defaultPort,
```

Wonder if we should throw instead if an invalid port is configured?

---

Refactored to move this to the new config style from https://github.com/n8n-io/n8n/pull/8446